sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-processing-steps/validate-disbursement-base.ts

```diff
@@ -0,0 +1,43 @@
+import { COEStatus } from "@sims/sims-db";
+import { ProcessSummary } from "@sims/utilities/logger";
+import { EligibleECertDisbursement } from "../disbursement-schedule.models";
+
+/**
+ * Common e-Cert validations for full-time and part-time.
+ */
+export abstract class ValidateDisbursementBase {
+  /**
+   * Validate common requirements independently of the offering intensity.
+   * @param eCertDisbursement eligible disbursement to be potentially added to an e-Cert.
+   * @param log cumulative log summary.
+   */
+  protected validate(
+    eCertDisbursement: EligibleECertDisbursement,
+    log: ProcessSummary,
+  ): boolean {
+    let shouldContinue = true;
+    // COE
+    if (eCertDisbursement.disbursement.coeStatus !== COEStatus.completed) {
```

Comment:
I more or less feel COEStatus = completed equivalent to application status = complete from the ecert perspective and could be in the query rather. 

Say if an application has 2 disbursements and is in original assessment, for the first COE the validation happens in query but for second COE we explicitly log it.

But no hard feeling against this way.